File: admin-web/src/api/prize.ts
import { http } from './http'

export interface Prize {
  id?: number
  type: string
  name: string
  description?: string
  handlerBean?: string
  paramsJson?: string
  totalStock?: number
  monthlyStock?: number
  dailyStock?: number
  userTotalLimit?: number
  userMonthlyLimit?: number
  userDailyLimit?: number
  limitsJson?: string
  activityId?: number
  groupKey?: string
  groupStrategy?: string
  groupWeight?: number
  iconUrl?: string
  claimZoneImageUrl?: string
  autoGrant?: boolean
  claimExpireType?: string
  claimExpireValue?: string
  maxRetry?: number
  enabled?: boolean
  startTime?: string
  endTime?: string
  createdAt?: string
  updatedAt?: string
}

export interface PrizeRecord {
  id?: number
  userId: string
  instanceId?: number
  taskId?: number
  stepId?: number
  prizeId?: number
  quantity?: number
  prizeType?: string
  prizeName?: string
  prizeIcon?: string
  prizeImage?: string
  status?: string
  expireTime?: string
  retryCount?: number
  errorMessage?: string
  externalTradeNo?: string
  wonAt?: string
  claimedAt?: string
  grantedAt?: string
}

export function listPrizes(page: number = 1, size: number = 20) {
  return http.get('/admin/prize', { params: { page, size } })
}

export function getPrize(id: number) {
  return http.get(`/admin/prize/${id}`)
}

export function createPrize(prize: Prize) {
  return http.post('/admin/prize', prize)
}

export function updatePrize(id: number, prize: Prize) {
  return http.put(`/admin/prize/${id}`, prize)
}

export function togglePrize(id: number) {
  return http.post(`/admin/prize/${id}/toggle`)
}

export function getPrizeRecords(id: number) {
  return http.get(`/admin/prize/${id}/records`)
}

File: backend/src/main/java/com/marketing/task/prize/domain/entity/Prize.java
package com.marketing.task.prize.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("prize")
public class Prize {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String type;
    private String name;
    private String description;
    private String handlerBean;
    private String paramsJson;
    private Integer totalStock;
    private Integer monthlyStock;
    private Integer dailyStock;
    private Integer userTotalLimit;
    private Integer userMonthlyLimit;
    private Integer userDailyLimit;
    private String limitsJson;
    private Long activityId;
    private String groupKey;
    private String groupStrategy;
    private Integer groupWeight;
    private String iconUrl;
    private String claimZoneImageUrl;
    private Boolean autoGrant;
    private String claimExpireType;
    private String claimExpireValue;
    private Integer maxRetry;
    private Boolean enabled;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String activityCode;
}
